collector/kernel/main.cc

```diff
@@ -641,6 +645,27 @@ int main(int argc, char *argv[])
 
     signal_manager.handle_signals({SIGINT, SIGTERM}, std::bind(&KernelCollector::on_close, &kernel_collector));
 
+    std::unique_ptr<scheduling::Timer> debug_bpf_lost_samples_timer;
+    if (schedule_bpf_lost_samples.given() && *schedule_bpf_lost_samples > 0) {
+      std::chrono::seconds const timeout(*schedule_bpf_lost_samples);
+
+      auto schedule_bpf_lost_samples = [&debug_bpf_lost_samples_timer, timeout]() {
+        if (auto const result = debug_bpf_lost_samples_timer->defer(timeout)) {
+          LOG::info("successfully scheduled inject_bpf_lost_samples() {} from now", timeout);
+        } else {
+          LOG::error("failed to schedule inject_bpf_lost_samples() {} from now: {}", timeout, result.error());
+        }
+      };
+
+      auto inject_bpf_lost_samples = [&kernel_collector, schedule_bpf_lost_samples]() {
+        kernel_collector.debug_bpf_lost_samples();
+        schedule_bpf_lost_samples();
+      };
+
+      debug_bpf_lost_samples_timer = std::make_unique<scheduling::Timer>(loop, inject_bpf_lost_samples);
+      schedule_bpf_lost_samples();
```

Comment:
Mm, mm, mm - layers of anonymous functions, how I love thee.  Perhaps some quick comments in here along the lines of 'schedule first interrupt,' 'reschedule interrupt after last fires,' etc.?  Just to make it a lot faster for the next person through this code.

Reply:
Will comment.

Reply:
Commented.